edx/edx-ora2
# A11y updates for ORA UI Studio Messaging

**TL;DR -** Collection of accessibility-related tweaks for the ORA editor views in Studio.

JIRA: [AU-91](https://openedx.atlassian.net/browse/AU-91) and [AU-92](https://openedx.atlassian.net/browse/AU-92)

**What changed?**

- Validation alert banner updates:
  - Add `aria-label=”warning”` to alert icon.
  - Add `role="alert"` to banner.
  - Add `aria-label=”close alert”` to close icon. Confirmed it is focusable and closable via keyboard.
- Tab updates:
  - Add `role="tab"`
  - Add screen reader text identifying the number of errors per tab.
  - Move warning icons to HTML elements (instead of pseudo-styling) and add `aria-label=””`
- Validation updates:
  - Add `aria-invalid="true"` to fields failing validation for each edit page.
- Updated some icon, text, and banner colors for better contrast (RE WCAG guidelines). See screenshots.
- Various refactors required to support the above.
- Unrelated doc updates. 

**Screenshots**
<img width="901" alt="ora-validation-icon-refresh" src="https://user-images.githubusercontent.com/12944465/131394503-d7600356-262c-49df-a71a-cce047c987e6.png">


**Developer Checklist**


**Testing Instructions**

1. In Studio > Course > ORA, click the "Edit" button to open the settings for an ORA.
2. For each tab (except "Prompt" which never fails validation):
  1. Introduce error(s) (see  https://github.com/edx/edx-ora2/pull/1662 for examples)
  2. Verify warning icons appear on tabs correctly
  3. In the browser inspection tools, verify:
     1. Screen-reader text shows number of errors.
     2. ARIA fields show invalid markup on fields failing validation
3. Verify other ARIA fields are correct (as listed in "What Changed?")

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:


FYI: @edx/masters-devs-gta
